waspc/src/Wasp/Generator/ServerGenerator/JobGenerator.hs

```diff
@@ -0,0 +1,76 @@
+module Wasp.Generator.ServerGenerator.JobGenerator
+  ( genJobs,
+    genJobFactories,
+  )
+where
+
+import Data.Aeson (object, (.=))
+import Data.Maybe
+  ( fromJust,
+  )
+import qualified GHC.Enum as Enum
+import StrongPath
+  ( Dir,
+    Path,
+    Posix,
+    Rel,
+    parseRelFile,
+    reldir,
+    reldirP,
+    relfile,
+    (</>),
+  )
+import Wasp.AppSpec (AppSpec, getJobs)
+import Wasp.AppSpec.Job (Job (perform))
+import Wasp.Generator.ExternalCodeGenerator.Common (GeneratedExternalCodeDir)
+import Wasp.Generator.FileDraft (FileDraft)
+import Wasp.Generator.JsImport (getJsImportDetailsForExtFnImport)
+import Wasp.Generator.Monad (Generator)
+import Wasp.Generator.ServerGenerator.Common
+  ( ServerSrcDir,
+  )
+import qualified Wasp.Generator.ServerGenerator.Common as C
+
+-- | TODO: Make this not hardcoded!
+relPosixPathFromJobFileToExtSrcDir :: Path Posix (Rel (Dir ServerSrcDir)) (Dir GeneratedExternalCodeDir)
+relPosixPathFromJobFileToExtSrcDir = [reldirP|../ext-src|]
+
+data JobFactory = PassthroughJobFactory
+  deriving (Show, Eq, Ord, Enum, Enum.Bounded)
+
+-- TODO: In future we will detect what type of JobFactory
+-- to use based on what the Job is using.
+jobFactoryForJob :: Job -> JobFactory
+jobFactoryForJob _ = PassthroughJobFactory
+
+genJobs :: AppSpec -> Generator [FileDraft]
+genJobs spec = return $ genJob <$> getJobs spec
+  where
+    tmplFile = C.asTmplFile [relfile|src/jobs/_jobs.js|]
+    dstFile jobName = C.asServerFile $ [reldir|src/jobs/|] </> fromJust (parseRelFile $ jobName ++ ".js")
+    genJob :: (String, Job) -> FileDraft
+    genJob (jobName, job) =
+      let (jobFnName, jobFnImportStatement) = getJsImportDetailsForExtFnImport relPosixPathFromJobFileToExtSrcDir $ perform job
+       in C.mkTmplFdWithDstAndData
+            tmplFile
+            (dstFile jobName)
+            ( Just $
+                object
+                  [ "jobName" .= jobName,
+                    "jobFnName" .= jobFnName,
+                    "jobFnImportStatement" .= jobFnImportStatement,
+                    "jobFactoryName" .= show (jobFactoryForJob job)
+                  ]
+            )
+
+genJobFactories :: Generator [FileDraft]
+genJobFactories = return $ genJobFactory <$> jobFactoryNames
+  where
+    genJobFactory :: String -> FileDraft
+    genJobFactory jobFactoryName =
+      let jobFactoryFile = [reldir|src/jobs/|] </> fromJust (parseRelFile $ jobFactoryName ++ ".js")
```

Comment:
Yeah, this does have a pretty strong assumption. I wasn't sure of the best way to handle it. The good news is it will fail pretty explicitly on any `compile` op like so:

```bash
🐝 --- Compiling wasp code... -----------------------------------------------------

wasp-cli: /home/shayne/dev/wasp/waspc/data/Generator/templates/server/src/jobs/AwesomeJobFactory.js: openBinaryFile: does not exist (No such file or directory)
```

So I think as a dev it would be pretty easy to spot and not slip through, but baking it into the TS somehow is certainly more ideal. What do you have in mind for the mapping function, something like this?

```haskell
jobFactoryFilePath :: JobFactory -> Path' (Rel d) File' 
jobFactoryFilePath PassthroughJobFactory = [relfile|src/jobs/PassthroughJobFactory.js|]
```

Yeah I think that may make it more clear at least 👍🏻 thnx

Reply:
Yeah, I agree. I like the hard coded/explicit so you are forced to handle any new factory cases, which should hopefully achieve the goal here. Thanks again for the suggestion!